### ISSUE TITLE
have to clean before changing build config

### ISSUE THREAD

**Issue by ioRekz** [emezeske/lein-cljsbuild#426]
When I'm done developing a feature and want to build the production js, I often forget to clean my outputs files so a "lein cljsbuild once min" for example would say "compiling" without actually building my app.js

Is there a way to force the build when I change the build config ? lein clean is not that hard but I would prefere not having to.

**Comment by mneise** [comment]
It should rebuild your project once you made changes to your code. Do you have a link to your project or could you attach your `project.clj` file?

**Comment by ioRekz** [comment]
Yes. But quite often i'm developing with lein figwheel. When everything is ok I start the prod build but my file has not changed so it build nothing.

**Comment by mneise** [comment]
Sorry, I can't reproduce your problem. I assume you are using different `output-dir`s in your config? It would be really helpful if you could post your config.

**Comment by magomimmo** [comment]
The lein cljsbuild clean subtask has been removed starting from lein-cljsbuild 1.0.4. So, even if you hook cljsbuild into the main leiningen tasks, when you call lein clean task, it only cleans the default lein :clean-targets.

https://github.com/technomancy/leiningen/blob/master/sample.project.clj#L305 <https://github.com/technomancy/leiningen/blob/master/sample.project.clj#L305>

HIH

**Comment by magomimmo** [comment]
sorry this is the right line number

https://github.com/technomancy/leiningen/blob/master/sample.project.clj#L303 <https://github.com/technomancy/leiningen/blob/master/sample.project.clj#L303>

**Comment by ioRekz** [comment]
Here is my file. https://gist.github.com/ioRekz/17ff586577cd930e69cb

Again, the problem is not the cleaning (which is actually working) but the fact that I have to clean or change a file to force a rebuild when I change the build config (from dev to min).

So I do a lein cljsbuild once dev.
Then lein cljsbuild once min will do nothing since nothing changed.

Because of that, I usually forgot to clean and upload my dev assets to my servers

**Comment by mneise** [comment]
What you could do is to have different `output-dir`s for your two different builds, for example:

```clojure
:cljsbuild {:builds [{:id "dev"
                      :figwheel true
                      :source-paths ["src"]
                      :compiler {:main foo.core
                                 :asset-path "js/out"
                                 :output-to "resources/public/js/out/foo.js"
                                 :output-dir "resources/public/js/out"
                                 :verbose true}}
                     {:id "min"
                      :source-paths ["src"]
                      :compiler {:main foo.core
                                 :asset-path "js/release"
                                 :output-to "resources/public/js/release/foo.js"
                                 :output-dir "resources/public/js/release"
                                 :optimizations :advanced}}]}
```

**Comment by ioRekz** [comment]
Indeed, but then I need 2 different index.html right ?

**Comment by mneise** [comment]
Yes.

**Comment by mneise** [comment]
This issue is the same as #414. Will look into fixing this with 2.0.
